### ISSUE TITLE
Mix audio from external stream

### ISSUE THREAD

**Issue by dbuezas** [AlexxIT/go2rtc#403]
Since I didn't manage to get the audio stream from the microphone, I made a local addon that runs this command:

```bash
ffmpeg -hide_banner -hwaccel vaapi -hwaccel_output_format vaapi -f v4l2 -video_size 640x480 -i /dev/video2 -c:v h264_vaapi -g 50 -bf 0 -profile:v high -level:v 4.1 -sei:v 0 -an -vf "format=vaapi|nv12,hwupload" -user_agent ffmpeg/go2rtc -rtsp_transport tcp -f rtsp rtsp://localhost:8554/webcam
```

And I have a  stream called `webcam` in the go2rtc config with only video. When I open it i the webrtcard, it magically picks up both video and audio. Is this the right way of doing it?
I also tried creating the final stream with both video and audio from my addon, but then I don't have where to send it too (since no stream is defined in go2rtc)

What would be the right way of doing this?

---

Btw, in my local addon I have these packages:
```dockerfile
RUN  apk add --no-cache \
        alsa-plugins-pulse=1.2.7.1-r0 \
        alsa-utils=1.2.8-r0 \
        pulseaudio-utils=16.1-r6 \
        ffmpeg 
```

and in the config.yaml, I added: 
```
audio: true
video: true
```

Maybe adding alsa and pulseaudio to the go2rtc docker container would allow for using the mic?

**Comment by dbuezas** [comment]
Here's the addon (it's extremely basic) https://github.com/dbuezas/microphone-stream-addon

**Comment by dbuezas** [comment]
(updated the ffmpeg command, I had pasted the wrong one)

**Comment by AlexxIT** [comment]
```
streams:
  camera1:
  - <some video source>
  - exec:ffmpeg -f alsa <other args> -rtsp_transport tcp -f rtsp {output}
```

**Comment by dbuezas** [comment]
Right, that would be perfect, but alsa doesn't work for me in the frigate nor go2rtc AddOns (because of missing packages I believe)

**Comment by AlexxIT** [comment]
Have you tried go2rtc hardware version (Debian linux)? Maybe it support alsa out of box. Looks like your example for Alpine.

**Comment by dbuezas** [comment]
I have not, I'll give it a try. Unfortunately frigate seems to crash often (and even crash the computer due to full ram) when using go2rtc 1.4.x
I'll retest this, maybe it depends on which streams i have or just a coincidence

**Comment by dbuezas** [comment]
I tried making a local add on with go2rtc so I could add the missing packages but I couldn't manage to get it to build the image (not fluent in docker). I think that addinig these should suffice:
```docker
RUN  apk add --no-cache \
        alsa-plugins-pulse=1.2.7.1-r0 \
        alsa-utils=1.2.8-r0 \
        pulseaudio-utils=16.1-r6 
```

**Comment by AlexxIT** [comment]
Was added to main and to hardware docker containers
https://github.com/AlexxIT/go2rtc/releases/tag/v1.5.0

**Comment by dbuezas** [comment]
It worked! 🎉 
```yaml
webcam:
    - ffmpeg:device?video=/dev/video2&input_format=yuyv422&video_size=640x480#video=h264#hardware
    - exec:ffmpeg -f alsa -ac 1 -i default -c:a libopus -ar:a 16000 -ac:a 1 -af adelay=0ms -rtsp_transport tcp -b:a 32k -listen 1 -reconnect 1 -f rtsp {output}
```

**Comment by dbuezas** [comment]
It's perfect!
I ended up dividing making a video only stream for frigate, and another one that copies is and adds audio for webrtc. This way there is no extra cpu usage from the microphone (which is surprisingly high) when it is not in use.
Thank you AlexxIT!

**Comment by dbuezas** [comment]
BTW, what is this? 
<img width="667" alt="image" src="https://user-images.githubusercontent.com/777196/236618074-c27adfa0-39e7-4793-8e12-28f4b5008d1e.png">

Is it an automatic suggestion from the add on for the microphone or is it detecting this from somewhere else that I may have misconfigured?

**Comment by dbuezas** [comment]
If this is coming from the add on as a suggestion for the microphone, I suggest to also add:
`ffmpeg -f alsa -ac 1 -i default -c:a libopus -ar:a 16000 -ac:a 1 -af adelay=0ms -rtsp_transport tcp -b:a 32k -listen 1 -reconnect 1 -f rtsp {output}`

libopus has much better latency (for me at least). The `adelay=0ms` makes the audio not choppy for some reason.
